STYLEGUIDE.md

```diff
@@ -35,6 +35,13 @@ If imports are shortened, the following conventions should be used. Full import
 paths are always acceptable.
 
 - `import probnum as pn`
+- `from probnum import randvars, linalg, diffeq, statespace`
+
+Please do not abbreviate import paths unnecessarily. We do **not** use the following imports:
+- `import probnum.random_variables as pnrv` or `import probnum.filtsmooth as pnfs` (correct would be `from probnum import randvars, filtsmooth`)
+- `from probnum import random_variables as rvs` or `import probnum.random_variables as rvs` (the `randvars` name is sufficiently short and does not need to be abbreviated)
+
+All of these rule obey the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html).
 
```

Comment:
There are two more factors I'd like to introduce here regarding imports, but I am not sure how subjective they are.

1. What about imports like `from probnum.diffeq import probsolve_ivp` in notebooks, for demonstration purposes? Currently this is featured quite heavily (especially in the linalg notebooks, if I remember correctly), but it deviates from the -- now -- fixed import style. I suggest we stick to `from probnum import diffeq` and avoid the other imports in notebooks as well. Thoughts? It does not need to be fixed yet, but I'd like to include it in the styleguide as a direction/
2. How about local relative imports. For instance, should the `_normal.py` import the base class as `from ._random_variables import RandomVariable`, or as `from probnum.randvars._random_variables import RandomVariable`, or as `from probnum import randvars` (which would introduce cyclic imports?) (there is more options, but I hope you get the question). Personally, I prefer importing the class `RandomVariable` directly here. I am not sure about the full path or a relative one in this case. Either way, this will deviate from the styleguide and I would like to write down a rule here.

On another note, shall I explicitly write down here that `typing` and `probnum.type` are excluded from the "full path" thing?

Reply:
ad 1) In my opinion importing functions in tutorials and docstrings is fine since the styleguide applies to the package itself, not how to use the package. However the NumPy and SciPy docstrings and tutorials do stick to exactly the style you suggest, so I will leave it up to you. If you want to put it in, I am all for it.

ad 2) I remember that there were a lot of issues with cyclic imports and having two instances of classes leading to strange bugs where `RandomVariables` were not instances of `RandomVariables`. Unfortunately I do not remember what the rule is that should be applied here.

ad 3) I think it is a good idea to explicitly write out the exceptions.

Reply:
Thanks for the feedback! 
1) I will make it all consistent then. I think it is better to do the same everywhere, and this way we are not forcing anyone to use the same style :) 
2) Then I'd say we leave it at the relative imports whenever modules "share a namespace" (you know what I mean). This is how it is currently done and does not seem to break anything. (Other than that, I prefer its readability anyway)
3) Alright!